src/pom.ts

```diff
@@ -1,17 +1,119 @@
-import { RequestHandler } from "express";
+import {RequestHandler} from "express";
+import {
+    CurseForgeError,
+    fetchAllModFiles,
+    fetchModFile,
+    fetchModMetadata,
+    isLatinLetter,
+    partition,
+    wasErroneous
+} from "./util";
+import {ModFileDependency, ModFileDependencyType, ModFileMetadata, ModMetadata} from "./modmetadata";
 
-const pom: RequestHandler = async (req, res) => {
-  const { descriptor, fileIds } = req.params
-  return res.send(
-    `<?xml version="1.0" encoding="UTF-8"?>
+export const generatePom = async (id: any, fileId: string, descriptor: string): Promise<string | CurseForgeError> => {
+    const metadata = await fetchModMetadata(id)
+    if (wasErroneous(metadata)) {
+        return metadata
+    }
+
+    const file = await fetchModFile(id, fileId)
+    if (wasErroneous(file)) {
+        return file
+    }
+    const publicationMillis = filePublicationMillis(file);
+    const dependencies = file.dependencies
+        .filter((it: ModFileDependency) =>
+            it.relationType === ModFileDependencyType.RequiredDependency ||
+            it.relationType === ModFileDependencyType.Include ||
+            it.relationType === ModFileDependencyType.Tool)
+        .map(async (dep: ModFileDependency) => await resolveDependencyFile(dep.modId + "", publicationMillis, file.gameId, file.gameVersions))
```

Comment:
As `resolveDependencyFile` is async, we don't need to have this function be async then immedietely await, as we can just return `resolveDependencyFile` as a promise immediately.

Also `dep.modId + ""` is a bit ugly, consider `String(dep.modId)` or \`${dep.modId}\`

You also don't need to explicitly state the type of the argument passed into the function `dep`, it should work it out for you!

```suggestion
        .map(dep => resolveDependencyFile(`${dep.modId}`, publicationMillis, file.gameId, file.gameVersions))
```

Reply:
Ah, ya that looks good, will commit it after I push.

Reply:
Nvm, didn't realize I can't do that... Pushed.